source/vir/src/ast_simplify.rs

```diff
@@ -436,6 +436,31 @@ fn simplify_one_expr(ctx: &GlobalCtx, state: &mut State, expr: &Expr) -> Result<
                 },
             ))
         }
+        ExprX::Assign { init_not_mut, lhs, rhs, op: Some(op) } => {
+            match &lhs.x {
+                ExprX::Var(_) | ExprX::VarLoc(_) => {
+                    Ok(
+                        SpannedTyped::new(
+                            &expr.span,
+                            &expr.typ,
+                            ExprX::Assign {
+                                init_not_mut: *init_not_mut,
+                                lhs: lhs.clone(),
+                                rhs: SpannedTyped::new(
+                                    &expr.span,
+                                    &lhs.typ,
+                                    ExprX::Binary(op.clone(), lhs.clone(), rhs.clone())
```

Comment:
It looks like you're cloning `lhs` as-is, but I think `lhs` might be a VarLoc, so should be changed to a `Var`

Reply:
done :+1: